airflow/providers/ftp/operators/ftp.py

```diff
@@ -0,0 +1,146 @@
+
+        Example: The following task would copy ``file.txt`` to the remote host
+        at ``/tmp/tmp1/tmp2/`` while creating ``tmp``,``tmp1`` and ``tmp2`` if they
+        don't exist. If the parameter is not passed it would error as the directory
+        does not exist. ::
+
+            put_file = FTPOperator(
+                task_id="test_ftp",
+                ftp_conn_id="ftp_default",
+                local_filepath="/tmp/file.txt",
+                remote_filepath="/tmp/tmp1/tmp2/file.txt",
+                operation="put",
+                create_intermediate_dirs=True,
+                dag=dag
+            )
+
+    """
+
+    template_fields: Sequence[str] = ('local_filepath', 'remote_filepath')
+
+    def __init__(
+        self,
+        *,
+        ftp_hook: FTPHook | None = None,
+        ftp_conn_id: str | None = None,
+        local_filepath: str | list[str],
+        remote_filepath: str | list[str],
+        operation: str = FTPOperation.PUT,
+        create_intermediate_dirs: bool = False,
+        **kwargs,
+    ) -> None:
+        super().__init__(**kwargs)
+        self.ftp_hook = ftp_hook
+        self.ftp_conn_id = ftp_conn_id
+        self.operation = operation
+        self.create_intermediate_dirs = create_intermediate_dirs
+
+        self.local_filepath_was_str = False
+        if isinstance(local_filepath, str):
+            self.local_filepath = [local_filepath]
+            self.local_filepath_was_str = True
+        else:
+            self.local_filepath = local_filepath
+
+        if isinstance(remote_filepath, str):
+            self.remote_filepath = [remote_filepath]
+        else:
+            self.remote_filepath = remote_filepath
+
+        if len(self.local_filepath) != len(self.remote_filepath):
+            raise ValueError(
+                f'{len(self.local_filepath)} paths in local_filepath '
+                f'!= {len(self.remote_filepath)} paths in remote_filepath'
+            )
+
+        if not (self.operation.lower() == FTPOperation.GET or self.operation.lower() == FTPOperation.PUT):
+            raise TypeError(
+                f"Unsupported operation value {self.operation}, "
+                f"expected {FTPOperation.GET} or {FTPOperation.PUT}."
+            )
+
+    def execute(self, context: Any) -> str | list[str] | None:
+        file_msg = None
+        try:
+            if self.ftp_conn_id and not self.ftp_hook:
+                self.log.info('ftp_hook not provided or invalid. Trying ftp_conn_id to create FTPHook.')
+                self.ftp_hook = FTPHook(ftp_conn_id=self.ftp_conn_id)
+
+            if self.ftp_hook and not isinstance(self.ftp_hook, FTPHook):
+                self.log.info('ftp_hook not provided or invalid. Trying ftp_conn_id to create FTPHook.')
+                self.ftp_hook = FTPHook(ftp_conn_id=self.ftp_conn_id)
+
+            if not self.ftp_hook:
+                raise AirflowException("Cannot operate without ftp_hook or ftp_conn_id.")
```

Comment:
Instead of all this I would created a method to get the hook using `self.ftp_conn_id`. I would also use @cached_property to get it only once

```suggestion
    @cached_property
    def hook(self) -> FTPHook:
        """Create and return an FTPHook."""
        return FTPHook(ftp_conn_id=self.ftp_conn_id)

    def execute(self, context: Any) -> list[str]:
        file_msg = None
        try:
            # You can now use the hook by just using `self.hook`
```